src/main/java/org/plp/implementations/plpisa/PlpFile.java

```diff
@@ -41,8 +43,14 @@ private void validateFilePath(Path filePath) throws IOException {
      * @return true if it is successful to write to file, false if there is an error
      */
     @Override
-    public boolean writeToFile() {
-        return false;
+    public boolean writeToFile() throws IOException{
+        try(BufferedWriter writer = Files.newBufferedWriter(filePath, Charset.forName("utf-8"))) {
```

Comment:
Need to check for empty file

Reply:
I think empty content to the file should be good. The assumption here would be we will be using writeToFile as a ctrl+s so sometimes there is a possibility that there is no change or not started writing the program or deleting the previous content of the file and saving it.